### PR TITLE
Make submission.reporting_round nullable

### PR DESCRIPTION
**Pull request by samuelhwilliams**
### Change description
Now that we're storing this information on
`programme_junction.reporting_round`, and we're no longer reading from `submission.reporting_round`, we can start to phase out writes to that table.

- [x] Unit tests and other appropriate tests added or updated
- [ ] README and other documentation has been updated / added (if needed)
- [x] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### How to test
_If manual testing is needed, give suggested testing steps_

### CODE REVIEW

File: db/migrations/versions/034_nullable_reporting_round.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+# revision identifiers, used by Alembic.
+revision = "034_nullable_reporting_round"
+down_revision = "033_project_geospatial_data"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    with op.batch_alter_table("submission_dim", schema=None) as batch_op:
```
Comment (wjrm500) [inline db/migrations/versions/034_nullable_reporting_round.py:20]: Hello again @samuelhwilliams just checking then, we are making this column nullable rather than dropping, or is there gonna be another PR for that?

Comment (samuelhwilliams) [reply db/migrations/versions/034_nullable_reporting_round.py:20]: We will drop it in the next PR :)

**Review by MarcUsher — APPROVED** [review]
Looks good - really helpful to see how you're splitting this up into smaller discrete PRs
